### ISSUE TITLE
Performance analysis of AEA and OEF interaction on R-PI

### ISSUE THREAD

**Issue by DavidMinarsch** [fetchai/agents-aea#1105]
**Is your feature request related to a problem? Please describe.**
There are issues with R-PI performance. Latency for OEF comms is extremely high.

**Describe the solution you'd like**
Identify what causes them and open issues for resolution

**Comment by DavidMinarsch** [comment]
> Hi @David M and @toby,
TL;DR
the issue on RPi is basically solved by the changes in the latest release re. the loop timeout.
---
I tested the code on a Raspberry Pi model B+, Raspbian 9.11 (codename: stretch). But the findings hold also for other models.
I've run an AEA on the RPi with the weather_client skill, connecting to an OEF node running on the same local network (my laptop). I tried both LOOP_TIMEOUT=0.0 (that reproduces the old behaviour) and LOOP_TIMEOUT=0.05. The screen recordings of the executions are timeout-0.mp4 and timeout-0.05.mp4, respectively.
As you can see, in timeout-0.mp4 the message is sent and the reply arrives, but the reply is not consumed. However, since timeout=0.0 , the control is never passed to the connection thread.
With timeout=0.05 instead, the reply is consumed relatively quickly.
So the issue was basically due to a framework configuration problem, and with the latest changes is now easy to configure that parameter.
